### PR TITLE
adding example simulating car IoT telemetry

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
Adding this new example

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -187,6 +187,26 @@ lazy val `java-eventing-shopping-cart` = project
...
+    libraryDependencies ++= Seq(
+        "ch.qos.logback" % "logback-classic" % LogbackVersion,
+        "com.novocode" % "junit-interface" % JUnitInterfaceVersion % IntegrationTest
+      ),
+    testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "-v", "-a"),
+    IntegrationTest / akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Java),
+    IntegrationTest / akkaGrpcGeneratedSources := Seq(AkkaGrpc.Client),
+    IntegrationTest / PB.protoSources ++= (Compile / PB.protoSources).value
+  )
+  .settings(attachProtobufDescriptorSets)
```
Comment (octonato) [inline build.sbt:208]: I'm wondering if there is any sbt plugin that we can use to delegate to mvn. 

This setup will work for simple projects, but we may run into issue when we start to deal dependencies in the project. We don't want to maintain deps here and in the maven. 

We may also run into other build things that we configure in the pom.xml and then again here. The gRPC setup is one of them, but we may have others.

**Review by octonato — COMMENTED** [review]
Looking good.

@franciscolopezsancho, I think we want to align the style with the codegen though. We want users to always be exposed to the same structure. 

Have you tried a project with the codegen?

File: samples/java-car-metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Registering Car metrics and selling data to Suppliers
+
+### Business logic
+A car emits data as an IoT device that we retrieve and then expose partially to suppliers. We provide the services for the cars to register its new data. Right now only the batery level (TODO add location and speed).
```
Comment (RayRoestenburg) [inline samples/java-car-metrics/README.md:4]:
```suggestion
A car emits data as an IoT device that we retrieve and then expose partially to suppliers. We provide the services for the cars to register its new data. Right now only the battery level (TODO add location and speed). 
```

File: samples/java-car-metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Registering Car metrics and selling data to Suppliers
+
+### Business logic
+A car emits data as an IoT device that we retrieve and then expose partially to suppliers. We provide the services for the cars to register its new data. Right now only the batery level (TODO add location and speed). 
+When new metrics are pushed through the API `car.metrics.api.CarMetricService/RecordChargeLevel` this data is not only recorded as an event in an EventSourcedEntity it is also forwarded to a Value Entity when the battery is lower than 25%. We represented this Value Entity as the data we sell and provide, through the API, to a supplier of batteries such only when the car is in LowBattery the supplier gains access to its data and it can act upon.
```
Comment (RayRoestenburg) [inline samples/java-car-metrics/README.md:5]:
```suggestion
When new metrics are pushed through the API `car.metrics.api.CarMetricService/RecordChargeLevel`, this data is not only recorded as an event in an EventSourcedEntity, it is also forwarded to a Value Entity when the battery charge level is lower than 25%. We represented this Value Entity as the data we sell and provide, through an API, to suppliers of batteries. The supplier only gains access to data when the car battery is low.   
```

File: build.sbt (hunk 1 above)
Comment (franciscolopezsancho) [reply build.sbt:208]: I think it is aligned with the other projects @octonato. I compared it to `java-eventing-shopping-cart` and seems identical to me.  Maybe I'm not understanding exactly what do you refer to with codegen.

Comment (octonato) [reply build.sbt:208]: Yep, it's aligned with the current status quo and that's the problem. 

We want all samples and docs to be based to the codegen. Have a look at https://github.com/lightbend/akkaserverless-java-sdk/pull/10 and https://github.com/lightbend/akkaserverless-java-sdk/pull/12

In the meantime, after many try and error and fighting with CircleCI, I managed to integrate the maven build for those two projects in the CI.

Comment (franciscolopezsancho) [reply build.sbt]: done now @octonato

Comment (octonato) [reply build.sbt]: @franciscolopezsancho, maybe you should rebase your PR with #10 and add a use the trick I added to build the maven with the recently built SDK version.

Comment (octonato) [reply build.sbt]: Then better to send your PR targeting the branch from #10, ie: https://github.com/lightbend/akkaserverless-java-sdk/tree/rgc/codegen-value-entity-sample.

We can then do cascade merging once we start to get them in `master`

**Comment by franciscolopezsancho** [comment]
@octonato  @ennru  shall I close this PR?

**Comment by ennru** [comment]
> shall I close this PR?

Maybe we can bring it over to lightbend-labs instead. WDYT @retgits ?

**Comment by retgits** [comment]
@ennru I think that's a good idea!

@franciscolopezsancho if you're okay with that too, let me know and I'll make sure you're invited to that org

**Comment by franciscolopezsancho** [comment]
Hi @retgits 
I can access https://github.com/lightbend-labs but I can't create a repo. Could you grant me permission for that, please?

**Comment by retgits** [comment]
Just to circle back here too, I added @franciscolopezsancho to `Lightbend-Labs` so he can create the repo :)

**Comment by franciscolopezsancho** [comment]
Done now 
https://github.com/lightbend-labs/akkaserverless-java-car-metrics

**Pull request closed without merging**